apps/antalmanac/src/components/dialogs/AddSchedule.tsx

```diff
@@ -0,0 +1,104 @@
+import { useCallback, useState, useEffect, useMemo } from 'react';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    TextField,
+    type DialogProps,
+    Box,
+} from '@mui/material';
+import { addSchedule } from '$actions/AppStoreActions';
+import { isDarkMode } from '$lib/helpers';
+import AppStore from '$stores/AppStore';
+
+type ScheduleNameDialogProps = DialogProps;
+
+/**
+ * Dialog with a text field to add a schedule.
+ */
+function AddScheduleDialog(props: ScheduleNameDialogProps) {
+    /**
+     * {@link props.onClose} also needs to be forwarded to the {@link Dialog} component.
+     * A custom {@link onKeyDown} handler is provided to handle the Enter and Escape keys.
+     */
+    const { onKeyDown, ...dialogProps } = props;
+
+    /**
+     * This is destructured separately for memoization.
+     */
+    const { onClose } = props;
+
+    const [scheduleNames, setScheduleNames] = useState(AppStore.getScheduleNames());
+
+    const defaultScheduleName = useMemo(() => `Schedule ${scheduleNames.length + 1}`, [scheduleNames]);
```

Comment:
I think the useState and useMemo aren't playing nice in `AddSchedule` since we're not rerendering the popover

![chrome-capture-2023-8-1 (2)](https://github.com/icssc/AntAlmanac/assets/100006999/f7ee52a6-4564-432d-aa4e-a4056b6052e3)